pkg/monitoring/services/services.go

```diff
@@ -0,0 +1,290 @@
+
+package services
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// ErrorNotImplementedForOS returned in case we don't yet implement service manager parsing or this OS. Should be checked and ignored
+var ErrorNotImplementedForOS = errors.New("Services list not implemented for " + runtime.GOOS)
+
+// SystemdService contains the service's data parsed from systemctl
+type SystemdService struct {
+	UnitFile    string
+	LoadState   string
+	ActiveState string
+	SubState    string
+	Description string
+}
+
+// SysVService contains the service's data parsed from service(Sysvinit) or initctl(Upstart)
+type SysVService struct {
+	UnitFile string
+	Status   string
+}
+
+// ListSystemdServices list Systemd services via systemctl
+func ListSystemdServices() ([]SystemdService, error) {
+	cmd := exec.Command("systemctl",
+		"--type=service", // show only services(ignore .mount, .target, .path, .socket etc.)
+		"--all",          // show loaded but inactive services too
+		"--no-pager",     // disable results pagination
+		"--plain",        // disable colors and status bullet
+		"list-units")
+	setPathEnvVar(cmd)
+
+	var outb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &outb
+	err := cmd.Run()
+	if err != nil {
+		errOutput, _ := ioutil.ReadAll(&outb)
+		return nil, fmt.Errorf("Systemctl: %s, %s", err.Error(), string(errOutput))
+	}
+
+	var services []SystemdService
+	scanner := bufio.NewScanner(&outb)
+	firstRow := true
+	var columns []string
+	for scanner.Scan() {
+		parts := strings.Fields(scanner.Text())
+
+		if firstRow {
+			// save columns to use it later
+			columns = parts
+
+			// When running `systemctl list-units` here in Golang it omits the last column's header "DESCRIPTION", but still writes the description columns with values
+			// Possibly it checks if stdout is terminal
+			// Workaround: add DESCRIPTION column if missed
+			if columns[len(columns)-1] != "DESCRIPTION" {
+				columns = append(columns, "DESCRIPTION")
+			}
+
+			firstRow = false
+			continue
+		}
+		if len(parts) == 0 {
+			// payload finished
+			// need to exit the read loop
+			break
+		}
+
+		rowColumnsValues := map[string]string{}
+		for colIndex, colName := range columns {
+
+			// most likely impossible but still need to check the boundaries
+			if colIndex >= len(parts) {
+				break
+			}
+
+			// if column is the last - join all the data left in the row because it can contain spaces
+			if colIndex == len(columns)-1 {
+				rowColumnsValues[colName] = strings.Join(parts[colIndex:], " ")
+				break
+			}
+
+			rowColumnsValues[colName] = parts[colIndex]
+		}
+
+		services = append(services,
+			SystemdService{UnitFile: rowColumnsValues["UNIT"], LoadState: rowColumnsValues["LOAD"], ActiveState: rowColumnsValues["ACTIVE"], SubState: rowColumnsValues["SUB"], Description: rowColumnsValues["DESCRIPTION"]},
+		)
+	}
+
+	return services, scanner.Err()
```

Comment:
we should check explicitly for error case and return `nil` if no error occurred.

Reply:
Mm... Like this? 
<img width="268" alt="image" src="https://user-images.githubusercontent.com/1025231/50515436-75979880-0aad-11e9-807c-311c44481497.png">

The problem is that `scanner.Error()` can theoretically return an error only after some successful so we already collected some services

Reply:
This is an error due to reading the output of intermediary-utility, so I think it may be reasonable to return both partial result and error in case it was failed.

Reply:
ok

Reply:
So.. Is it OK to leave it as is if we keep partial results?